### PR TITLE
Laboratory work #3, Olga Kostina - 23FPL2

### PR DESCRIPTION
**Pull request by olyakostii1na**

### CODE REVIEW

File: lab_3_ann_retriever/main.py [hunk 1]
```diff
@@ -118,6 +129,23 @@ def tokenize_documents(self, documents: list[str]) -> list[list[str]] | None:
...
+                or not documents:
+            return None
+
+        tokenized_documents = []
+
+        for document in documents:
+            tokenized_document = self.tokenize(document)
+            if tokenized_document is None:
+                return None
+            document_without_stopwords = self._remove_stop_words(tokenized_document)
```
Comment (Vasilisa-Blyudova) [inline lab_3_ann_retriever/main.py:142]: do you really need to use `_remove_stop_words` method?

File: lab_3_ann_retriever/main.py [hunk 2]
```diff
@@ -131,6 +159,17 @@ def _remove_stop_words(self, tokens: list[str]) -> list[str] | None:
...
         """
 
+        if not isinstance(tokens, list) or not all(isinstance(token, str) for token in tokens) or not tokens:
+            return None
+
+        clear_text = []
+
+        for token in tokens:
+            if token not in self._stop_words:
+                clear_text.append(token)
```
Comment (Vasilisa-Blyudova) [inline lab_3_ann_retriever/main.py:169]: try to use list comprehension

**Review by Vasilisa-Blyudova — CHANGES_REQUESTED** [review]

**Comment by Vasilisa-Blyudova** [comment]
No progress since last review

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
No new changes

**Pull request closed without merging**